### ISSUE TITLE
Get lucee 'deploy' directory via CFconfig for purpose of extension installation

### ISSUE THREAD

**Issue by mborn319** [Ortus-Solutions/cfconfig#10]
How can I install the `<cfspreadsheet>` Lucee extension during a Bitbucket Pipelines build?

According to the [extension docs](https://github.com/Leftbower/cfspreadsheet-lucee-5), you can install the extension by simply putting the `.lex` file in the Lucee server's `deploy/` folder.

> Download and copy the cfspreadsheet-lucee-5.lex into the /lucee-server/deploy or {lucee-web}/deploy directory of a running Lucee 5 installation.

Here's what I use on my Mac for setting up a development environment:

    curl https://raw.githubusercontent.com/Leftbower/cfspreadsheet-lucee-5/master/cfspreadsheet-lucee-5.lex -O
    cp cfspreadsheet-lucee-5.lex ~/.CommandBox/server/*-CMS4Dev/lucee-5.2.4.37/WEB-INF/lucee-server/deploy

These two commands in Bash work perfectly fine **on my local development server**, but in Bitbucket Pipelines it doesn't seem to find the Lucee installation directory. I didn't expect it to either - I'd prefer accessing the `lucee-server/deploy/` or `lucee-web/deploy/` folders from [CFconfig](https://github.com/Ortus-Solutions/cfconfig), but I cannot find any mention of this.

To recap: Add a feature or method by which we can retrieve the `deploy/` directory for a web or server context. If we could get the directory path, it's easy to programmatically install the extension via simple shell commands.

**Comment by bdw429s** [comment]
Hi @mborn319 , the correct place for tickets in CFConfig is actually here:
https://ortussolutions.atlassian.net/projects/CFCONFIG/issues
Looks like I need to disable Github issues.

As for your request, I think it might be ill-placed.  It's not CFConfig's job to tell you anything about your server.  In fact, CFConfig requires _you_ to tell it where the server lives (unless, of course, it's a CommandBox server in which case the CFConfig CLI will be able to figure it out).

So a few questions here:
1. Are you actually using the CFConfig services directly, or are you using the CFConfig CommandBox CLI, which is in this repo here:  https://github.com/Ortus-Solutions/commandbox-cfconfig
2. Are you using CFConfig with a CommandBox server, or a typical Tomcat-based installer-produced installation?  Your script above seems to show a CommandBox server, but I'm unclear on whether that was for your local machine, for Bitbucket, or both.
3. If you simply want to know where the server home is for a CommandBox-based server, then CFConfig cannot tell you, but CommandBox can.  Run 
```
server info --json
```
and you'll see a whole bunch of information that CommandBox can tell you about a server.  You can easily build up the path to a given server by getting the server home property like so:
```
server info property=serverHomeDirectory
```
Then just append `/WEB-INF/lucee-server/deploy` to it (assuming you haven't overridden the `serverConfigDir` to be elsewhere.  We're most likely not going to add any dedicated functions or properties for one-off folders like deploy since there's a 1000 different folders in there that someone _might_ want, but so long as you have the server home dir, then you can build out the rest of the path using the known file structure.  

And on a totally different note, is the cfspreadsheet extension part of the built-in Lucee extension provider?  Ifr so, there an even easier way you can load it in by specifying some JVM args, but it's a bit more of a pain if it requires a 3rd party extension provider.  I have yet to add the ability to CFConfig to add in 3rd party extension providers.

**Comment by mborn319** [comment]
Hi Brad. 
Sorry for putting this issue in the completely wrong repo!

1. Nope, I'm using the CLI. My mistake.
2. This is on a CommandBox server. I'm hoping/assuming I can make this extension installation work for both local development and Bitbucket Pipelines.
3. `server info property=serverHomeDirectory` is perfect and is exactly what I need. Thank you!

Unfortunately it is not in any extension provider, as far as I know. I don't know **that** much about providers, but I know the Lucee server admin can't find any "spreadsheet" extension when you try to search/install this from the GUI.

*"Then just append /WEB-INF/lucee-server/deploy to it..."*
How would I do this? I would expect this line to work:
```
echo "`server info property=serverHomeDirectory`/WEB-INF/lucee-server/deploy/cfspreadsheet-lucee-5.lex" | mv cfspreadsheet-lucee-5.lex
```
but you can't pipe a filename to either `mv` or `cp`, they both take standard input as the destination instead of source argument. Can I stick the server directory in a variable somehow? Then the `mv` command becomes as easy as 
```
mv $extFile $serverDir
````

Script
```
# STEP ONE: Download the file
run curl https://raw.githubusercontent.com/Leftbower/cfspreadsheet-lucee-5/master/cfspreadsheet-lucee-5.lex -O

# STEP TWO: Move to the "deploy" directory for the server context.
echo "`server info property=serverHomeDirectory`/WEB-INF/lucee-server/deploy/cfspreadsheet-lucee-5.lex" | mv cfspreadsheet-lucee-5.lex
```

**Comment by bdw429s** [comment]
If you're looking to do the dynamic `mv` in bash, then just look up how to do that in bash.  Expansions can be inline though for parameters so you should be able to do something like this instead of using `echo`:
```
$> mv my.lex "`box server info property=serverHomeDirectory`/WEB-INF/lucee-server/deploy/"
```
I didn't actually test that.  

And yes you can set variables in bash.  This is the wrong place to ask about Bash usage though, you should be able to google that anywhere.  (Note, you need `box` in the expansion too).

Now, if you're asking about doing the copy from within CommandBox, that's possible too, but the syntax might be a bit different from bash.
